swift/Sources/FlatBuffers/FlatBuffersBuilder.swift

```diff
@@ -0,0 +1,385 @@
+import Foundation
+
+//// TODO: - add docs
+public final class FlatBuffersBuilder {
+    
+    private var _vtable: [UInt32] = []
+    private var _vtables: [UOffset] = []
+    private var _bb: FlatBuffer
+    private var isNested = false
+    private var _numOfFields = 0
+    private var stringOffsetMap: [String: Offset<String>] = [:]
+    private var finished = false
+    private var serializeDefaults: Bool
+    private var objectStart = 0
+    
+    var size: UOffset { return _bb.size }
+    var _minAlignment: Int32 = 0 {
+        didSet {
+            _bb.alignment = Int(_minAlignment)
+        }
+    }
+    
+    public var data: Data {
+        if !finished { fatalError(FlatbufferError.calledSizedBinaryBeforeFinish.localizedDescription) }
+        return Data(bytes: _bb.memory, count: _bb.capacity)
+    }
+    
+    public var fullSizedByteArray: [UInt8] {
+        let ptr = UnsafeBufferPointer(start: _bb.memory.assumingMemoryBound(to: UInt8.self), count: _bb.capacity)
+        return Array(ptr)
+    }
+    
+    public var sizedByteArray: [UInt8] {
+        let cp = _bb.capacity - _bb.writerIndex
+        let ptr = UnsafeBufferPointer(start: _bb.memory.advanced(by: _bb.writerIndex).bindMemory(to: UInt8.self, capacity: cp), count: cp)
+        return Array(ptr)
+    }
+    
+    public var buffer: FlatBuffer { return _bb }
+    
+    // MARK: - Init
+    
+    ///
+    /// - Parameter initialSize:
+    public init(initialSize: Int32 = 1024, serializeDefaults force: Bool = false) {
+        guard initialSize > 0 else { fatalError(FlatbufferError.sizeIsZeroOrLess.errorDescription ?? "") }
+        guard isLitteEndian else { fatalError(FlatbufferError.endianCheck.errorDescription ?? "") }
```

Comment:
> * Looking at the docs this is not clear, but I am guessing that `UnsafeMutableRawPointer` when it reads a 16/32/64-bit quantity, it will do so with machine-endianness ordering, which means it won't work correctly on big endian machines. This means you should either have explicit big-endian versions of most of these functions, or (and this is entirely acceptable) explicitly disable this code to run on big-endian.

Currently, indeed the code is disabled from running on Big-endian machines. In hopes that we would be implementing it whenever we get to have an issue regarding it.

Reply:
That's fine.